### ISSUE TITLE
Replace AsClientOptions, AsClientMethodOptions with derive macro-defined functions

### ISSUE THREAD

**Issue by heaths** [Azure/azure-sdk-for-rust#1737]
Once the derive macro is complete for client and client method options, we should remove the `AsClientOptions` and `AsClientMethodOptions` added for #1734 as a stopgap solution. They'd serve no purpose at that point and just end up being yet another import devs will have to add; though, rust-analyzer should make short work of that.

**Comment by heaths** [comment]
No longer relevant since we're using PODO
